src/main/java/frc/robot/subsystems/CoralArm.java

```diff
@@ -0,0 +1,51 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotMap;
+
+public class CoralArm extends SubsystemBase {
+    private DoubleSolenoid piston;
+    private DoubleSolenoid piston2;
+    private DigitalInput extendlimitswitch;
+    private DigitalInput retractlimitswitch;
+    public CoralArm(){
+        this.piston = new DoubleSolenoid(PneumaticsModuleType.REVPH, RobotMap.PISTON_FORWARD_CHANNEL,RobotMap.PISTON_BACKWARD_CHANNEL);
+        this.piston2 = new DoubleSolenoid(PneumaticsModuleType.REVPH, RobotMap.PISTON2_FORWARD_CHANNEL,RobotMap.PISTON2_BACKWARD_CHANNEL);
+        this.extendlimitswitch = new DigitalInput(1);
+        this.retractlimitswitch = new DigitalInput(2);
```

Comment:
```suggestion
        this.retractlimitswitch = new DigitalInput(RobotMap.CORAL_ARM_RETRACTED_SWITCH);
```
Ports should be defined in `RobotMap`